github/github.go

```diff
@@ -567,6 +613,17 @@ func (c *Client) RateLimits() (*RateLimits, *Response, error) {
 		return nil, nil, err
 	}
 
+	if response.Resources != nil {
+		c.rateMu.Lock()
+		if response.Resources.Core != nil {
```

Comment:
Yeah, that sounds weird... not sure what I meant by that... I guess it would just be nice to see a test case for one or both of these.  Your call.